src/common/sha2/sha2_ossl.c

```diff
@@ -50,20 +54,21 @@ void OQS_SHA2_sha256_inc_init(OQS_SHA2_sha256_ctx *state) {
 	md = oqs_sha256();
 	OQS_EXIT_IF_NULLPTR(md);
 	mdctx = EVP_MD_CTX_new();
-	EVP_DigestInit_ex(mdctx, md, NULL);
+	OQS_EXIT_IF_NULLPTR(mdctx);
+	OQS_OPENSSL_GUARD(EVP_DigestInit_ex(mdctx, md, NULL));
 	state->ctx = mdctx;
 }
 
 void OQS_SHA2_sha256_inc_blocks(OQS_SHA2_sha256_ctx *state, const uint8_t *in, size_t inblocks) {
-	EVP_DigestUpdate((EVP_MD_CTX *) state->ctx, in, inblocks * SHA2_BLOCK_SIZE);
+	OQS_OPENSSL_GUARD(EVP_DigestUpdate((EVP_MD_CTX *) state->ctx, in, inblocks * SHA2_BLOCK_SIZE));
 }
 
 void OQS_SHA2_sha256_inc_finalize(uint8_t *out, OQS_SHA2_sha256_ctx *state, const uint8_t *in, size_t inlen) {
 	unsigned int md_len;
 	if (inlen > 0) {
-		EVP_DigestUpdate((EVP_MD_CTX *) state->ctx, in, inlen);
+		OQS_OPENSSL_GUARD(EVP_DigestUpdate((EVP_MD_CTX *) state->ctx, in, inlen));
 	}
-	EVP_DigestFinal_ex((EVP_MD_CTX *) state->ctx, out, &md_len);
+	OQS_OPENSSL_GUARD(EVP_DigestFinal_ex((EVP_MD_CTX *) state->ctx, out, &md_len));
 	EVP_MD_CTX_free((EVP_MD_CTX *) state->ctx);
```

Comment:
Would there be any value in setting `state->ctx = NULL` after this as a precaution against double frees?

Reply:
so changed in https://github.com/open-quantum-safe/liboqs/pull/1454/commits/5f8d02c9c2651b9a5f3eca0364457a5b271a473a